### PR TITLE
vaultwarden.webvault: 2024.1.0 -> 2024.1.1

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

Didn't deploy it yet but the changes are minimal, so it should continue to work.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/security/vaultwarden/webvault.nix [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
+  bw_web_builds = applyPatches {
+    src = fetchFromGitHub {
+      owner = "dani-garcia";
+      repo = "bw_web_builds";
+      rev = "v${version}";
+      hash = "sha256-xtfpxcJLP0C4FdnO45gsaecOWJ/cKC++Abm7iatTH1Y=";
+    };
+
+    patches = [
+      #  Fix applying latest patch if patches is a symlink
```
Comment (dotlambda) [inline pkgs/tools/security/vaultwarden/webvault.nix:23]:
```suggestion
      # Fix applying latest patch if patches is a symlink
```

File: pkgs/tools/security/vaultwarden/webvault.nix [hunk 2]
```diff
@@ -23,10 +37,10 @@ in buildNpmPackage rec {
...
     rev = "web-v${lib.removeSuffix "b" version}";
-    hash = "sha256-lDDy1b1yfw3nZrwEEkpvh6xYucgn20XHsGACc45eb2w=";
+    hash = "sha256-695iCkFhPEyyI4ekbjsdWpxgPy+bX392/X30HyL4F4Y=";
   };
 
-  npmDepsHash = "sha256-RR8Ua41D9SXymiPuabOnIab3byu8DR63rOfdeTaQpy4=";
+  npmDepsHash = "sha256-IJ5JVz9hHu3NOzFJAyzfhsMfPQgYQGntDEDuBMI/iZc=";
 
   postPatch = ''
     ln -s ${bw_web_builds}/{patches,resources} ..
```
Comment (dotlambda) [inline pkgs/tools/security/vaultwarden/webvault.nix:46]: We could also do 
```suggestion
    cp -r ${bw_web_builds}/{patches,resources} ..
```
to fix this.

Comment (SuperSandro2000) [reply pkgs/tools/security/vaultwarden/webvault.nix]: Good idea. My PR got also accepted, so with the next update it doesn't matter anymore.

**Comment by dotlambda** [comment]
@ofborg test vaultwarden

**Comment by SuperSandro2000** [comment]
Selenium doesn't work on aarch64, hence the test failure.

**Comment by dotlambda** [comment]
> Selenium doesn't work on aarch64

Is there an issue about that?

**Review by dotlambda — APPROVED** [review]
LGTM once the commit message is fixed.

**Comment by SuperSandro2000** [comment]
> Selenium

Not that specifically but there are two that certain files are missing. They are using bazel, so I won't try to look into that again.

**Pull request merged**